### PR TITLE
Extract plugin version from manifest

### PR DESCRIPTION
**Pull request by Hanzei**
This allows for e.g. log messenger when the plugin starts with the current version.

I have no experience in js. Should similar changes be made to `plugin_id.js`?

### CODE REVIEW

File: build/manifest/main.go [hunk 1]
```diff
@@ -31,7 +32,10 @@ func main() {
 	cmd := os.Args[1]
 	switch cmd {
 	case "plugin_id":
```
Comment (Hanzei) [inline build/manifest/main.go]: I think this should be changed to "id" to be a simpler command. 4/5

File: build/manifest/main.go [hunk 2]
```diff
@@ -76,17 +80,22 @@ func findManifest() (*model.Manifest, error) {
 	return &manifest, nil
 }
 
-// dumpPluginId writes the plugin id from the given manifest to standard out
-func dumpPluginId(manifest *model.Manifest) {
+// dumpPluginID writes the plugin id from the given manifest to standard out
+func dumpPluginID(manifest *model.Manifest) {
```
Comment (jwilander) [inline build/manifest/main.go]: Can we leave it as `dumpPluginId`? I know some style checkers complain about it but this is how we do it all over and we try to be consistent

**Comment by Hanzei** [comment]
Thanks for the review! I understand that you want to keep things consistent. If we follow this thought we should use one file for both web and server e.g. `manifest.js` and `manifest.go` or two files for both web and server e.g. `plugin_version.js` and `plugin_id.js` and `plugin_id. go` and `plugin_version.go`. 

I would rather go with one file per apppart.

File: build/manifest/main.go (hunk 2 above)
Comment (lieut-data) [reply build/manifest/main.go]: 0/5 for this PR, but I do not this isn't just about "some style checkers" but more about writing idiomatic Golang, e.g. https://github.com/golang/go/wiki/CodeReviewComments#initialisms

**Comment by lieut-data** [comment]
@Hanzei, I think a generated `manifest.(js|go)` probably makes a lot of sense!

**Comment by jwilander** [comment]
@Hanzei would you mind copying these changes to https://github.com/mattermost/mattermost-plugin-demo as well?

**Comment by Hanzei** [comment]
@jwilander sure, I can do this

**Comment by Hanzei** [comment]
See https://github.com/mattermost/mattermost-plugin-demo/pull/6

**Pull request merged**